DashVtk examples

RohitRathore1 opened the pull request:
Started to work on DashVtk:
1. Currently added about DashVtk and what we will cover
2. Working on Intro to 3D Visualization
3. Set up the all paths and default view

RohitRathore1 committed:
DashDaq work

RohitRathore1 committed:
intro about 3D Visualization

mohdibntarek commented on DashDocs/src/chapters/dash_vtk/IntroToVisualization/examples/index.jl:
````diff
@@ -0,0 +1,109 @@
+
+    ```
+    dash_vtk.View(
+      id='vtk-view',
````
This is not Julia syntax. Please make sure this is runnable Julia code.

RohitRathore1 replied:
Yes, I am looking into it and will keep in mind further. Thanks for the review.

mohdibntarek commented on DashDocs/src/chapters/dash_vtk/IntroToVisualization/examples/index.jl:
```diff
@@ -0,0 +1,109 @@
+
+    A mouse event can be identified with the following set of properties:
+    - *button*: 1/2/3 # Which button should be down
+    - *shift*: True/False # Is the `Shift` key down
```
True/False is not Julia syntax, I think it should be true/false

mohdibntarek commented:
There seems to be a conflict with master. Please resolve it.

RohitRathore1 commented:
> There seems to be a conflict with master. Please resolve it.

Ok, Sure.

RohitRathore1 committed:
Merge branch 'master' into dash_examples

RohitRathore1 committed:
added one advanced demo example

RohitRathore1 committed:
Merge branch 'dash_examples' of https://github.com/RohitRathore1/dash-julia-docs into dash_examples

RohitRathore1 commented:
Started to work on DashVtk:
1. Currently added about DashVtk and what we will cover
2. Working on Intro to 3D Visualization
3. Set up all paths and default view

Update:
Now, we have an example from Advanced Demos called Synthetic Volume Rendering which is related to medical and fixed the issues from previous reviews.

mohdibntarek commented on DashDocs/src/chapters/dash_vtk/IntroToVisualization/examples/index.jl:
````diff
@@ -0,0 +1,109 @@
+    For the *interactorSettings* we expect a list of mouse event type linked to an action. The example below is what is used as default:
+    ```
+    interactorSettings=[
+      {
````
This should be a Julia Dict

RohitRathore1 commented:
Can we go for the implementation of [utils.py](https://github.com/plotly/dash-vtk/blob/master/dash_vtk/utils/vtk.py) in Julia because we have defined some functions which are required for the conversion of examples and the defined functions are: `to_mesh_state`, `to_volume_state`, `present_as_options`.?

RohitRathore1 committed:
work on structure of datasets

RohitRathore1 commented:
Started to work on DashVtk:
1. Currently added about DashVtk and what we will cover
2. Working on Intro to 3D Visualization
3. Set up all paths and default view

Update:
Now, we have an example from Advanced Demos called Synthetic Volume Rendering which is related to medical and fixed the issues from previous reviews. 

Update:
In Structure of Dataset now we have converted examples of:
a) vtk_imagedata
b) vtk_polydata

RohitRathore1 committed:
added another section

RohitRathore1 commented:
Started to work on DashVtk:

Currently added about DashVtk and what we will cover
Working on Intro to 3D Visualization
Set up all paths and default view
Update:
Now, we have an example from Advanced Demos called Synthetic Volume Rendering which is related to medical and fixed the issues from previous reviews.

Update:
In Structure of Dataset now we have converted examples of:
a) vtk_imagedata
b) vtk_polydata

Update:
Now, we have a complete section of
1. Structure Dataset
2. Representation Components

mohdibntarek commented on DashDocs/src/app/default_page.jl:
```diff
@@ -88,9 +88,9 @@ function default_page()
                     "The Dash DAQ library contains a set of higher-level components like boolean switch, color picker, gauge and more."
                 ),
                 Chapter(
-                    "Dash DataTable",
```
don't remove this

mohdibntarek commented on DashDocs/src/DashDocs.jl:
```diff
@@ -15,7 +15,7 @@ include("chapters/dash_core_components/index.jl")
 include("chapters/dash_html_components/index.jl")
 include("chapters/dash_daq/index.jl")
 include("chapters/dash_bio/index.jl")
-include("chapters/dash_data_table/index.jl");
```
don't remove this, just add both

RohitRathore1 committed:
fixed after getting review

RohitRathore1 committed:
fixed eror

mohdibntarek commented on DashDocs/src/chapters/dash_vtk/RepresentationComponents/index.jl:
```diff
@@ -0,0 +1,351 @@
+        function vtk_pointcloudrepresentation(; kwargs...) 
+                 return vtk_geometryrepresentation(
+                     id=kwargs.get("id"),
+                     colorMapPreset=kwargs.get("colorMapPreset"),
```
Using kwargs.get like that is not idiomatic in Julia. You probably want to do `kwargs[:colorMapPreset]`

mohdibntarek replied:
Please address this in a separate PR.
